ambassador/tests/test_ambassador.py

```diff
@@ -709,6 +709,27 @@ def check(self):
                 actual = lowercased_headers.get(k.lower())
                 assert actual == [v], "expected %s: %s but got %s" % (k, v, lowercased_headers)
 
+class RemoveResponseHeaders(OptionTest):
+
+    parent: Test
+
+    VALUES: ClassVar[Sequence[Dict[str, str]]] = (
+        { "foo": "bar" },
+        { "moo": "arf" }
+    )
+
+    def config(self):
+        yield "remove_response_headers: %s" % "x-envoy-upstream-service-time"
+
+    def check(self):
+        for r in self.parent.results:
+            # Why do we end up with capitalized headers anyway??
+            lowercased_headers = { k.lower(): v for k, v in r.headers.items() }
+
+            for k, v in self.value.items():
```

Comment:
I agree that having this as an `OptionTest` is great, but this doesn't seem to actually test that the `X-Envoy-Upstream-Service-Time` header was actually removed?

Reply:
Hah, lol. I can't run these and seems like I totally forgot to finish it when I came back to push it. I revamped the test in latest commit, but still couldn't run it so not sure if it passes